### ISSUE TITLE
[ui] UI app does not work with CSRF tokens

### ISSUE THREAD

**Issue by sduenas** [chaoss/grimoirelab-sortinghat#316]
When running the app, the server raises a **403 Forbidden** error due the CSRF token is not set. The app can work if we add this code to the `urls.py` file:

```python
from django.contrib import admin
from django.urls import path

from sortinghat.core.views import SortingHatGraphQLView

from .schema import schema

from django.views.decorators.csrf import csrf_exempt


urlpatterns = [
    path('admin/', admin.site.urls),
    path('graphql/', csrf_exempt(SortingHatGraphQLView.as_view(graphiql=True, schema=schema))),
]
```

That disables the CSFR checkings and makes the server vulnerable. To fix it, the app client should include CSRF tokens on their queries to the SortingHat client. The https://github.com/apollographql/apollo-client/issues/132 issue contains useful information, especially https://github.com/apollographql/apollo-client/issues/132#issuecomment-532865657.

**Comment by dlumbrer** [comment]
Hi @sduenas,

I tried to solve the problem using the solution proposed in the ticket, but unfortunately, it didn't work for me. I still have the CSRF error, as follows:

![Screenshot from 2020-09-21 16-08-15](https://user-images.githubusercontent.com/9249232/93777175-b5f55d80-fc24-11ea-9a52-efa9a9a914e4.png)

I did some research about the problem and I reached this question (https://stackoverflow.com/questions/56241554/django-graphene-apollo-django-webpack-loader-vue-cors-csrf-not-working-together) with a very similar scenario that ours, but it does not have an answer or explanation.

**Comment by sduenas** [comment]
@dlumbrer, in the command line client, what we do is establish an initial connection with the server using `GET` method. The response contains a `csrf` token. WIth that token, we set the headers and we are able to make `POST` requests (call to GraphQL methods).

I don't know much about Apollo but maybe we can try the same method. I think the solution proposed on the issue doesn't get the token from the beginning.

**Comment by dlumbrer** [comment]
I prepared a PR (#328) solving this Issue, It cost me a lot of time to do it, summarizing the code, we have to retrieve the Cookie after forcing a first GET petition to the Django, and then add it to the Apollo server/client configuration.

I encountered a problem because I had to activate the flag "credentials" in the creation of the httpLink (https://github.com/chaoss/grimoirelab-sortinghat/pull/328/files#diff-70080f29ee72e10a229d118a8900edf5R23), and it was difficult to understand and find it.

Please @sduenas, check it when you can

**Comment by sduenas** [comment]
Resolved by #328. The only drawback is we need to access the web client using `localhost` instead of  `127.0.0.1`. We'll try to fix this in future versions.
